libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -1860,6 +1867,61 @@ namespace
    }
 }
 
+auto WaveTrack::FindWideClip(const WaveClip &clip, int *pDistance) const
+   -> ConstIterPair
+{
+   const WaveClipHolders &list = mClips;
+   const WaveClipHolders *pList2{};
+   const auto channels = TrackList::Channels(this);
+   if (channels.size() > 1)
+      pList2 = &(*channels.rbegin())->mClips;
+   int distance = 0;
+   auto it = list.begin();
+   for (const auto end = list.end(); it != end; ++it) {
+      if (it->get() == &clip)
+         break;
+      ++distance;
+   }
+   if (pDistance)
+      *pDistance = distance;
+   WaveClipHolders::const_iterator it2{};
+   if (pList2)
+      it2 = pList2->begin() + distance;
```

Comment:
This assumes that the insertion order of clips in each of the left- and right channel tracks is kept the same. What makes you confident that it is? Did you consider an approach comparing metadata (e.g. clip offset and length?)

Reply:
I'll do a bounds check